python/src/equistore/operations/join.py

```diff
@@ -17,6 +17,40 @@ def join(tensormaps: List[TensorMap], axis: str):
     properties of the TensorMaps and if ``axis='samples'`` it will be the
     along the samples.
 
+    Possible clashes of the meta data like the property names
+    are resolved by the following strategy:
+
+    1. If the names (i.e property names) are the same and the
+       values (i.e property values)
+       are unique we keep the names and
+       join the values vertically. In example::
+
+        labels_1 = ["n"], [0, 2, 3]
+        labels_2 = ["n"], [1, 4, 5]
+
+      will lead to::
+
+        properties = ["n"], [0, 2, 3, 1, 4, 5]
+
+    2. If the names are the same but the values are not unique a new variable
+       "tensor" is added to the names. In example::
+
+            properties_1 = ["n"], [0, 2, 3]
+            properties_2 = ["n"], [0, 2]
+
+        will lead to::
+            properties = ["tensor", "n"], [[0, 0], [0, 2], [0, 3], [1, 0], [1, 2]]
```

Comment:
wouldn't it be better if the property is called `tensor_n`, so
```suggestion
            properties = ["tensor_n", "n"], [[0, 0], [0, 2], [2, 0], [2, 2], [1, 2]]
```
to differ it from other properties that are resolved the same way? Especially because you cannot have multiple times the same name

Reply:
If this is more a core issue about the join function, we can also move the discussion to an issue to merge this fix.

Reply:
Yes, I agree. I think we should merge this fix asap. We should however try to improve the joining of labels further.